### PR TITLE
Proc entry stat pr

### PR DESCRIPTION
**Pull request by emkey1**
I know this is present in an issue, but I'm trying to do things correctly and this is a small and easy test case.  

I now have a branch in my fork that will track the true master.  So it should be relatively easy for me to generate clean pull requests.

I also plan on submitting only one thing at a time rather than multiples.

### CODE REVIEW

**Comment by emkey1** [comment]
Built from current master and using the default root fs, but with an iSH user added for testing.

Screen Shot 2020-12-23 at 1 42 07 PM

**Comment by tbodt** [comment]
> I also plan on submitting only one thing at a time rather than multiples.

Good idea, that way a problem with one thing won't hold up submitting other things.

**Comment by tbodt** [comment]
Could you get rid of the merge commit and base this branch directly on tbodt master?

**Review by tbodt — COMMENTED** [review]
This isn't the best separation of concerns, but it can always be fixed later

**Comment by emkey1** [comment]
> Could you get rid of the merge commit and base this branch directly on ish-app:master?

I'll see what I can do.  It took a lot of research and a small miracle to get to this point.  Basically I created a branch master, rolled back everything that had happened since I created my branch and then did a pull request from your current master to get my master back in sync with yours.  

Mucking up my master branch was really really dumb in retrospect but I have learned a lot trying to extricate myself from that mess.  Not enough though it seems.

File: fs/proc/entry.c [hunk 1]
```diff
@@ -21,6 +22,18 @@ mode_t_ proc_entry_mode(struct proc_entry *entry) {
 int proc_entry_stat(struct proc_entry *entry, struct statbuf *stat) {
     memset(stat, 0, sizeof(*stat));
     stat->mode = proc_entry_mode(entry);
+    
+    lock(&pids_lock);
+    struct task *task = pid_get_task(entry->pid);
+    if (task != NULL) {
+       stat->uid = task->uid;
```
[the remark this answers is not in the document]
Comment (emkey1) [reply fs/proc/entry.c:29]: Fixed in pending commit

File: fs/proc/entry.c [hunk 2]
```diff
@@ -21,6 +22,18 @@ mode_t_ proc_entry_mode(struct proc_entry *entry) {
...
+       stat->gid = task->gid;
+    } else {
+        stat->uid = 0;
+        stat->gid = 0;
```
[the remark this answers is not in the document]
Comment (emkey1) [reply fs/proc/entry.c:33]: Good point.  I'll leave a comment to this effect and git rid of the else

**Comment by emkey1** [comment]
> Could you get rid of the merge commit and base this branch directly on ish-app:master?

I've tried a few things, but short of closing this one and generating a new pull request I haven't been able to do that.  

It seems like it should be simple, but I get the following when I try to revert...

```
error: commit a0eb16d22ad9783d7e5adb32585cf258a9eb4919 is a merge but no -m option was given.
fatal: revert failed
```

Searching online provided no joy.  It's particularly annoying in that the commit is essentially empty so it seems like this should be easy?

**Review by wachabaha — APPROVED** [review]

File: fs/proc/entry.c [hunk 3]
```diff
@@ -21,6 +22,17 @@ mode_t_ proc_entry_mode(struct proc_entry *entry) {
...
+    lock(&pids_lock);
+    struct task *task = pid_get_task(entry->pid);
+    
+    if (task != NULL) {
+        stat->uid = task->uid;
+        stat->gid = task->gid;
+    } // else the memset above will have initialized memory to zero, which is the root uid/gid
+    
+    unlock(&pids_lock);
+
```
Comment (ahmadkoresh4) [inline fs/proc/entry.c:35]: ahmadkoresh4@gmail.com

**Comment by ahmadkoresh4** [comment]
> Built from current master and using the default root fs, but with an iSH user added for testing.
> 
> Screen Shot 2020-12-23 at 1 42 07 PM

**Comment by tbodt** [comment]
thanks for the contribution! sorry it took so long to merge

**Comment by emkey1** [comment]
> thanks for the contribution! sorry it took so long to merge

Cool.  Thank you!

Now I need to get stuff cleaned up on my end so I can submit some of the other things I have pending.

**Pull request merged**